yarn-project/simulator/src/avm/errors.ts

```diff
@@ -39,6 +40,37 @@ export class InvalidProgramCounterError extends AvmExecutionError {
   }
 }
 
+/**
+ * Error is thrown when the program counter points to a byte
+ * of an invalid opcode.
+ */
+export class InvalidOpcodeError extends AvmExecutionError {
+  constructor(opcode: Opcode) {
```

Comment:
Let's only have a string here, and format the message on the caller. Then we avoid the dependency on "Opcode".